File: app/Models/Play.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Play extends Model
{
    protected $fillable = [
        'song_isrc',
        'seconds_played',
    ];

    public function song()
    {
        return $this->belongsTo(Song::class, 'song_isrc', 'isrc');
    }
}

File: app/models/play.rb
class Play < ApplicationRecord
  # belongs_to is required by default, which also enforces that the ISRC
  # exists in songs (the Laravel app validated exists:songs,isrc).
  belongs_to :song, foreign_key: :song_isrc, inverse_of: :plays

  validates :seconds_played, numericality: { only_integer: true, greater_than_or_equal_to: 1 }
end
